bgpd/bgp_mplsvpn.c

```diff
@@ -233,6 +236,762 @@ int bgp_nlri_parse_vpn(struct peer *peer, struct attr *attr,
 #undef VPN_PREFIXLEN_MIN_BYTES
 }
 
+static int ecom_intersect(struct ecommunity *e1, struct ecommunity *e2)
+{
+	int i;
+	int j;
+
+	if (!e1 || !e2)
+		return 0;
+
+	for (i = 0; i < e1->size; ++i) {
+		for (j = 0; j < e2->size; ++j) {
+			if (!memcmp(e1->val + (i * ECOMMUNITY_SIZE),
+				    e2->val + (j * ECOMMUNITY_SIZE),
+				    ECOMMUNITY_SIZE)) {
+
+				return 1;
+			}
+		}
+	}
+	return 0;
+}
+
+static void leak_update(struct bgp *bgp, /* destination bgp instance */
+			struct bgp_node *bn,
+			struct attr *new_attr, /* already interned */
+			afi_t afi, safi_t safi, struct bgp_info *source_bi,
+			u_char type, u_char sub_type, uint32_t *label,
+			void *parent, int debug)
+{
+	struct prefix *p = &bn->p;
+	struct bgp_info *bi;
+	struct bgp_info *new;
+
+	/*
+	 * match peer, parent
+	 */
+	for (bi = bn->info; bi; bi = bi->next) {
+		if (bi->peer == source_bi->peer && bi->extra
+		    && (struct bgp_info *)bi->extra->parent == parent) {
+
+			break;
+		}
+	}
+
+	if (bi) {
+		if (attrhash_cmp(bi->attr, new_attr)
+		    && !CHECK_FLAG(bi->flags, BGP_INFO_REMOVED)) {
+
+			bgp_attr_unintern(&new_attr);
+			if (debug)
+				zlog_debug("%s: Found route, no change",
```

Comment:
Can we have a more specific debug statement ? This is pretty generic and won't give any information while debugging. Or if this is not necessary, can we please remove it ?

Reply:
some additional information like the prefix at the least ? generally we will grep for a specific  prefix or some attr while debugging. It will be helpful if we print that information too in the same line.

Reply:
It sounds like a good idea. Let me look at these more closely...